vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -269,9 +269,7 @@ protected void fireSelectionEvent(
                     @Override
                     public void setDeselectAllowed(boolean deselectAllowed) {
                         super.setDeselectAllowed(deselectAllowed);
-                        grid.getElement().executeJs(
-                                "this.$connector.deselectAllowed = $0",
-                                deselectAllowed);
+                        grid.updateDeselectAllowed(deselectAllowed);
```

Comment:
You could probably use `grid.getElement().setProperty('deselectAllowed', deselectAllowed)` and and then make the `gridConnector` read `grid.deselectAllowed` instead of `grid.$connector.deselectAllowed`. Then Flow will take case of synchronizing the state on attach.

Reply:
Thanks, seems to work. Done.